tools/apiKey.js

```diff
@@ -0,0 +1,55 @@
+// coding=utf-8
+
+// Copyright [2024] [SkywardAI]
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//        http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { API_KEY_TABLE } from "../database/types.js";
+import { getTable } from "../database/index.js";
+
+const MAX_USAGE = 10;
+const tbl = await getTable(API_KEY_TABLE);
+
+function queryApiKeyTbl(key) {
+    const keyQuery = tbl.query()
+        .where('api_key = '+"'"+key+"'")
+        .limit(1)
+        .toArray();
+    return keyQuery
+}
+
+export async function test(key){
```

Comment:
test functions should be removed

Reply:
Sorry - Just updated that. Also I was not sure how we wanted to implement error handling.

Reply:
the idea is that, if the validation is not passed or usage runs out, then the api sends corrsponding status and tells the reason, for example `res.status(401).send("User not authorized")` or `res.status(403).send("Out of usage")`